mondrian/src/it/java/mondrian/test/LocalizingDynamicSchemaProcessorTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+* This software is subject to the terms of the Eclipse Public License v1.0
+* Agreement, available at the following URL:
+* http://www.eclipse.org/legal/epl-v10.html.
+* You must accept the terms of that agreement to use this software.
+*
+* Copyright (c) 2019 Hitachi Vantara..  All rights reserved.
+*/
+
+package mondrian.test;
+
+import junit.framework.TestCase;
+import mondrian.i18n.LocalizingDynamicSchemaProcessor;
+import mondrian.olap.MondrianProperties;
+import mondrian.olap.Util;
+import mondrian.spi.DynamicSchemaProcessor;
+
+/**
+ * Unit test LocalizingDynamicSchemaProcessor. Tests availability of properties that LDSP's
+ * are called, and used to modify the resulting Mondrian schema
+ *
+ * @author bcosta
+ */
+public class LocalizingDynamicSchemaProcessorTest
+    extends TestCase
+{
+    /**
+     * Tests to make sure that our LocalizingDynamicSchemaProcessor works, with
+     * replacements. Does not test Mondrian is able to connect with the schema
+     * definition.
+     */
+    public void testProcessingCatalog() {
+        String variable = "%{translate}";
+        String translation = "Translate";
+        String catalog = "..." + variable + "...";
+        MondrianProperties.instance().LocalePropFile.set( "resources.properties" );
```

Comment:
Two things here:

 - Is there a way not to use external files? I don't particularly like the idea of an external test file. MAybe refactor a method into something you could override in the next line below when you create the LDSP. If not, at least put the file in a proper folder, not at the root of the test resources.

- The property needs to be reset at the end of this particular test method. Normally, all of our test are subclasses of FoodMartTest, which exposes the propSaver protected object. Setting properties through the propSaver will automatically reset them after the test. Might be worth looking into.

Reply:
- I am going put a file in a different folder closer to class to be tested. I think that using a properties file inside a resource file (external file as you called it) is the better approach because it allow us to test loading file content to our object.

- I based my test on DynamicSchemaProcessorTest.java.